R/ml_fit.R

```diff
@@ -48,6 +54,17 @@ ml_fit <- function(ml_problem,
     verbose = verbose, ...)
 }
 
+#' @rdname ml_fit
+#' @export
+ml_fit.list <- function(ml_problem,
```

Comment:
If all this code is doing is calling `lapply()`, do we really need it? I think it can be expected from users to iterate over lists if needed?

Reply:
I proposed this because it can nicely work in pipe workflows regardless of which kind of fitting_problem the user creates. Any other downsides to this design that I may not be aware of?

Reply:
We sacrifice type stability of `ml_fit()` for a relatively minor gain. For pipe workflows could we use:

```r
. %>%
  map(ml_fit) %>%
  ...
```

?

Reply:
I see. Ultimately, which solutions would you prefer? let the user iterates the list by themself or make zoneless problems become a list of length 1 internally (a842db99a0bd89f3b0d6997268dad23988161517)?

Reply:
Both work. If we let the user iterate, they get a bit more control, but we need to document the pattern. OTOH, they can always control by creating sub-problems manually.

Do you have a preference?

Have you seen the dm package? Would it be reasonable to provide a converter from a suitable dm object to a fitting problem?

Reply:
I think for maintainability, less code is always better. :) I'm cherry picking commits from this PR and will open a new PR that only have the `geo_hierarchy` feature.

Yes, I have used the dm package. That sounds like a good feature to have. However, right now I would like to focus more on releasing the package to CRAN and the paper, since both of these tasks have been delayed quite a bit now. :(

Will keep the dm feature as a feature request and will work on it in a future release.